pkg/kubelet/kubelet.go

```diff
@@ -1159,6 +1192,7 @@ func (kl *Kubelet) SyncPods(allPods []*api.Pod, podSyncTypes map[types.UID]metri
 	//      though the pod was not considered terminated by the apiserver).
 	// These two conditions could be alleviated by checkpointing kubelet.
 	pods := kl.filterOutTerminatedPods(allPods)
+	pods = kl.filterOutPodsPastActiveDeadline(pods)
```

Comment:
I guess what I find confusing about the current structure is the following:

Step 1: handleNotFittingPods
The input list of pods to this function is all pods bound to the Kubelet.  It checks the pod.Spec.Containers to check for host port conflicts.  If a pod is failed, it does not make sense that it should be taken into account for host port conflicts at all. If there is a conflict though, we fail a pod, and throw an event.

Step 2: Filter out the terminated pods

Step 3: Filter out the pods past active deadline
Since we already have a list of pods that were not terminated, we do not get called twice for the same pod.  This stops duplicate events.

... REPEAT ...

On the subsequent repeat, we throw a host port conflict again because we do not ignore terminated pods for not fitting pods.  That seems like a bug in the current structure.